msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -148,23 +179,33 @@ public async Task<IEnumerable<IAccount>> GetAccountsAsync()
         }
 
         /// <summary>
-        /// Get the account by identifier from the accounts available in the cache.
+        /// Get the <see cref="IAccount"/> by its identifier among the accounts available in the token cache.
         /// </summary>
-        /// <param name="accountId">account identifier</param>
+        /// <param name="accountId">account identifier. The value of the identifier will probably have been stored value from the
+        /// value of the <see cref="AccountId.Identifier"/> property of <see cref="AccountId"/>. 
+        /// You typically get the account id from an <see cref="IAccount"/> by using the <see cref="IAccount.HomeAccountId"/> property>
+        /// </param>
         public async Task<IAccount> GetAccountAsync(string accountId)
         {
             var accounts = await GetAccountsAsync().ConfigureAwait(false);
             return accounts.FirstOrDefault(account => account.HomeAccountId.Identifier.Equals(accountId, StringComparison.OrdinalIgnoreCase));
         }
 
         /// <summary>
-        /// Attempts to acquire the access token from cache. Access token is considered a match if it AT LEAST contains all the requested scopes.
-        /// This means that an access token with more scopes than requested could be returned as well. If access token is expired or 
-        /// close to expiration (within 5 minute window), then refresh token (if available) is used to acquire a new access token by making a network call.
+        /// Attempts to acquire and access token for the <paramref name="account"/> from the user token cache. 
         /// </summary> 
-        /// <param name="scopes">Array of scopes requested for resource</param>
-        /// <param name="account">User for which the token is requested. <see cref="IAccount"/></param>
-        /// <returns></returns>
+        /// <param name="scopes">scopes requested to access a protected API</param>
+        /// <param name="account">Account for which the token is requested. <see cref="IAccount"/></param>
+        /// <returns>An <see cref="AuthenticationResult"/> containing the requested token</returns>
+        /// <exception cref="MsalUiRequiredException">can be thrown in the case where an interaction is required with the end user of the application, 
+        /// for instance so that the user consents, or re-signs-in (for instance if the password expirred), or performs two factor authentication</exception>
+        /// <remarks>
+        /// The access token is considered a match if it AT LEAST contains all the requested scopes.
```

Comment:
>AT LEAST [](start = 57, length = 8)

remove? then perhaps add what else we are looking for besides scopes? This seems to be a topic which will come up a lot in the future... where we may end up failing the request if the scopes returned by STS doesn't match what we sent... I think it deserves it's own aka.ms link (explaining more about scopes and how we and STS handles them?)

Reply:
I've added a work item for that.